pkg/controller/kafkachannel/dispatcher.go

```diff
@@ -252,14 +253,14 @@ func (r *Reconciler) newDispatcherDeployment(channel *kafkav1beta1.KafkaChannel)
 					},
 				},
 				Spec: corev1.PodSpec{
-					ServiceAccountName: r.environment.ServiceAccount,
+					ServiceAccountName: r.config.ServiceAccount,
 					Containers: []corev1.Container{
```

Comment:
yeah, again here i don't see why we can't continue to use the environment (if that's where they are) instead of artificially adding to the config ?